assets/chat/js/menus/ChatUserInfoMenu.js

```diff
@@ -66,6 +46,32 @@ export default class ChatUserInfoMenu extends ChatMenu {
     });
   }
 
+  showUser(e, user, userlist = false) {
+    this.clickedNick = user.data('username');
+
+    this.setActionsVisibility();
+    this.addContent(user, userlist);
+
+    const rect = this.chat.output[0].getBoundingClientRect();
+    // calculating floating window location (if it doesn't fit on screen, adjusting it a bit so it does)
+    const x =
+      this.ui.width() + e.clientX > rect.width
+        ? e.clientX - rect.left + (rect.width - (this.ui.width() + e.clientX))
+        : e.clientX - rect.left;
+    const y =
+      this.ui.height() + e.clientY > rect.height
+        ? e.clientY -
+          rect.top +
+          (rect.height - (this.ui.height() + e.clientY)) -
+          12
+        : e.clientY - rect.top - 12;
+
+    this.ui[0].style.left = `${x}px`;
+    this.ui[0].style.top = `${y}px`;
```

Comment:
Now that we have two menus with floating windows (including `ChatEmoteTooltip` from #95), I think this positioning logic should be moved somewhere else. Maybe another class that both menus extend. Can you merge master back into your branch and take care of this, please?

Reply:
@11k Done.